apps/designer/app/designer/features/style-panel/sections/typography/typography.tsx

```diff
@@ -1,19 +1,318 @@
+  TextStrikethroughIcon,
+  TextTruncateIcon,
+  TextUnderlineIcon,
+  TextUppercaseIcon,
+} from "@webstudio-is/icons";
+import { ToggleGroupControl } from "../../controls/toggle/toggle-control";
+import { ValuePickerPopover } from "../../shared/value-picker-popover";
 
-export const TypographySection = ({
-  styleConfigsByCategory,
-  moreStyleConfigsByCategory,
-}: RenderCategoryProps) => {
+export const TypographySection = (props: RenderCategoryProps) => {
   return (
-    <>
-      {styleConfigsByCategory.map((entry) => renderProperty(entry))}
-      <ShowMore
-        styleConfigs={moreStyleConfigsByCategory.map((entry) =>
-          renderProperty(entry)
-        )}
-      />
-    </>
+    <Flex css={{ gap: "$spacing$7" }} direction="column">
+      <TypographySectionFont {...props} />
+      <TypographySectionSizing {...props} />
+      <TypographySectionAdvanced {...props} />
+    </Flex>
   );
 };
+
+export const TypographySectionFont = (props: RenderCategoryProps) => {
+  const { sectionStyle } = props;
+
+  return (
+    <Grid
+      css={{
+        gap: "$spacing$5",
+      }}
+    >
+      <Grid css={{ gridTemplateColumns: "4fr 6fr" }}>
+        <PropertyName label="Font" property="fontFamily" />
+        <FontFamilyControl {...sectionStyle.fontFamily} />
+      </Grid>
+      <Grid css={{ gridTemplateColumns: "4fr 6fr" }}>
+        <PropertyName label="Weight" property="fontWeight" />
+        <FontWeightControl {...sectionStyle.fontWeight} />
+      </Grid>
+      <Grid css={{ gridTemplateColumns: "4fr 6fr" }}>
+        <PropertyName label="Color" property="color" />
+        <ColorControl {...sectionStyle.color} />
+      </Grid>
+    </Grid>
+  );
+};
+
+export const TypographySectionSizing = (props: RenderCategoryProps) => {
+  const { sectionStyle } = props;
+
+  return (
+    <Grid
+      css={{
+        gridTemplateColumns: "1fr 1fr 1fr",
+        gap: "$spacing$5",
+      }}
+    >
+      <Grid css={{ gridTemplateColumns: "auto", gap: "$spacing$3" }}>
+        <PropertyName property="fontSize" label="Size" />
+        <TextControl {...sectionStyle.fontSize} />
+      </Grid>
+      <Grid css={{ gridTemplateColumns: "auto", gap: "$spacing$3" }}>
+        <PropertyName property="lineHeight" label="Height" />
+        <TextControl {...sectionStyle.lineHeight} />
+      </Grid>
+      <Grid css={{ gridTemplateColumns: "auto", gap: "$spacing$3" }}>
+        <PropertyName property="letterSpacing" label="Spacing" />
+        <TextControl {...sectionStyle.letterSpacing} />
+      </Grid>
+    </Grid>
+  );
+};
+
+export const TypographySectionAdvanced = (props: RenderCategoryProps) => {
+  const { setProperty, currentStyle } = props;
+  const setTextAlign = setProperty("textAlign");
+  const setTextDecorationLine = setProperty("textDecorationLine");
+  const setTextTransform = setProperty("textTransform");
+  const setFontStyle = setProperty("fontStyle");
+
+  return (
+    <Grid
+      css={{
+        gap: "$spacing$5",
```

Comment:
same

Reply:
please check all of them

Reply:
not sure how to feel about usefullness of the Grid component if we have to override with CSS prop this much, lets have a look at how others do it, e.g. material-ui etc, maybe we could add more props

Reply:
I consider every css override I see a bad thing, it means it's some edge case or bad component design